staging/src/k8s.io/apiserver/pkg/admission/plugin/cel/interface.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cel
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/apiserver/pkg/cel"
+
+	"github.com/google/cel-go/common/types/ref"
+)
+
+type FailurePolicy string
+
+const (
+	Fail   FailurePolicy = "Fail"
+	Ignore FailurePolicy = "Ignore"
+)
+
+type EvaluatorFunc func(a admission.Attributes, params *unstructured.Unstructured) []PolicyDecision
+
+// Dependency Injected into the PolicyDefinition's `Compile` function to assist
+// with converting types and values to/from CEL-typed values
+type ObjectConverter interface {
+	// Given a GVR, look up the openapi or JSONSchemaProps, structural schema, etc.
+	// and compile it into something that can be used to turn objects into CEL
+	// values
+	DeclForResource(gvr schema.GroupVersionResource) *cel.DeclType
```

Comment:
@cici37 please advise whether we want GVR or GVK as argument to this.

Reply:
Probably you want GVR.

Reply:
Also, there should be a returned error when, for example, the GVR cannot be found, especially when looking up CRDs.

Reply:
> Is there a way to look for the GVK given the GVR?

I think that is outside of the scope of this interface. Whatever is implementing it can use whatever tools it needs to complete the job.

> Also, there should be a returned error when, for example, the GVR cannot be found, especially when looking up CRDs.

added